ci/generate-and-upload-build-steps.sh

```diff
@@ -1,15 +1,73 @@
 #!/bin/bash
 set -euo pipefail
 
+upload_build_configuration_step() {
```

Comment:
General comment on shell style: 

Prefer variable expansion like `"${MY_VAR}"` to `"$my_var"` (its a little safer & forces a consistent style through the file)

Reply:
Also this script feels like its _almost_ at the complexity at which it shouldn't be a shell script anymore

Reply:
Agreed - fixed the variable style now. If any extra logic has to be added in I'll migrate to some proper langauge that can run on ths build capability 😛

Reply:
My personal preference would be to convert this to a e.g. C# project. This script is already pretty complex (but very well documented!) and we will need to add MacOS & mobile CI at some point (most likely in the next quarter). It will just make this script even more complex, but if we already move to something else it will be easier to add those new build targets in the future.

Reply:
@ImprobableVlad could you please create a task to track this refactoring/redesign? 
Please add Jessica and Jamie comments as supporting points to the effort. thanks

Reply:
@michelefabris As suggested by you, I've created a ticket to track progress on this issue: https://improbableio.atlassian.net/browse/GV-156

Reply:
thanks! :D